src/AbstractSyntaxTreeNode.ts

```diff
@@ -8,41 +7,30 @@ import BaseTestRunner from './testrunners/BaseTestRunner';
  * @constructor
  */
 export default class AbstractSyntaxTreeNode {
+  private _node: ESTree.Node;
```

Comment:
I see this pattern a lot. I can agree that you want to isolate the properties (and thus not making them writable), but i would still like to see parameter properties. That would save a lot of boilerplate.